core/src/test/scala/org/apache/spark/scheduler/TaskSetManagerSuite.scala

```diff
@@ -316,6 +316,36 @@ class TaskSetManagerSuite extends SparkFunSuite with LocalSparkContext with Logg
     assert(manager.resourceOffer("exec2", "host2", ANY).get.index === 3)
   }
 
+  test("Scheduler respects maxRunningTasks setting of its pool") {
+    sc = new SparkContext("local", "test")
+    val sched = new FakeTaskScheduler(sc, ("exec1", "host1"))
+    val taskSet = FakeTask.createTaskSet(5)
+    val clock = new ManualClock
+    val manager = new TaskSetManager(sched, taskSet, MAX_TASK_FAILURES, clock)
+    manager.parent = new Pool("pool", SchedulingMode.FAIR, 0, 2, 0)
+
+    // pool has two tasks...
+    assert(manager.resourceOffer("exec1", "host1", ANY).get.index === 0)
+    assert(manager.resourceOffer("exec1", "host1", ANY).get.index === 1)
+
+    // but not three!
+    assert(manager.resourceOffer("exec1", "host1", ANY).isEmpty)
+
```

Comment:
The test case should include scheduling another taskset to a different pool which does not share the limitation, and making sure it can still schedule tasks even when the first task set gets limited.